Функциональность «Рекомендации»

Phronima6 opened the pull request:
Добавить функциональность «Рекомендации»

Commit:
Функциональность «Рекомендации»

LadyMoonMare reviewed (approved):
Хорошая работа, Артём! Замечаний не имею.
Единственное, на будущее - советую в дальнейшей работе избегать нагромождения комментариев, это затрудняет чтение кода.

zolotyh-dk commented on src/main/java/ru/yandex/practicum/filmorate/service/user/UserServiceImpl.java:
```diff
@@ -83,4 +86,68 @@ public List<User> getCommonFriends(Integer id, Integer otherId) {
+        }
+        return likeStorage.getFilmLikes(candidateId).stream()
+                // Фильтруем поток, исключая фильмы, которые лайкнул пользователь
+                .filter(film -> !likeStorage.getFilmLikes(userId).contains(film))
```
`likeStorage.getFilmLikes(userId)` возможно будет лучше это запросить один раз в самом начале метода и сохранить в переменной, чтобы не обращаться к БД каждый раз внутри стрима.

Phronima6 replied:
Спасибо! Ты полностью прав, это сэкономит ресурс!

zolotyh-dk reviewed (commented):
Стримы как у синьора!

Commit:
Привести в соответствие с зачеманиями:
- в классе UserServiceImpl, в методе getRecommendations добавить хранение списков понравившихся пользователю фильмов в List<Film> для избежания повторных обращений к базе данных и экономии ресурсов;
- убрать лишние комментации для лучшей четаемости кода.
